### PR TITLE
Adding More Data Points for telemetry

### PR DESCRIPTION
**Pull request by khilangoel**

### CODE REVIEW

File: src/Microsoft.TestPlatform.Common/Telemetry/TelemetryDataConstants.cs [hunk 1]
```diff
@@ -31,6 +31,22 @@ public static class TelemetryDataConstants
...
+
+        public static string MaxCPUcount = "VS.TestRun.MaxCPUcount";
+
+        public static string TestPlatformVersion = "VS.TestRun.TestPlatformVersion";
+
+        public static string TargetOS = "VS.TestRun.TargetOS";
+
+        public static string LoggerUsed = "VS.TestRun.LoggersUsed";
+
+        public static string AdaptersVersion = "VS.TestRun.AdaptersVersion";
```
[the remark this answers is not in the document]
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.Common/Telemetry/TelemetryDataConstants.cs]: Will remove it. It is not needed. Talked with Vinod.

File: src/Microsoft.TestPlatform.Common/Telemetry/TelemetryDataConstants.cs [hunk 2]
```diff
@@ -44,9 +60,6 @@ public static class TelemetryDataConstants
 
         public static string NumberOfAdapterUsedToRunTests = "VS.TestRun.AdaptersUsedCount";
 
-        // In case of parallel, it will be maximum of all time taken by different execution engine.
-        public static string TimeTakenToStartExecutionEngineExe = "VS.TestRun.ExecutionEngineStartTime";
```
[the remark this answers is not in the document]
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.Common/Telemetry/TelemetryDataConstants.cs]: Do we need this?

File: src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs [hunk 3]
```diff
@@ -189,6 +195,8 @@ public void InitializeLoggers()
                     }
                 }
             }
+
+            requestData.MetricsCollection.Add(TelemetryDataConstants.LoggerUsed, string.Join(",", loggersUsed.ToArray()));
```
[the remark this answers is not in the document]
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs]: In both cases URI will be logged.

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs [hunk 4]
```diff
@@ -280,5 +294,36 @@ private static string UpdateExtensionsFolderInRunSettings(string settingsXml)
                 return document.OuterXml;
             }
         }
+
+        /// <summary>
+        /// Collect Enabled Data Collectors
+        /// </summary>
+        private void CollectEnabledDataCollectors()
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs]: Suggestion : Rename it

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs [hunk 5]
```diff
@@ -280,5 +294,36 @@ private static string UpdateExtensionsFolderInRunSettings(string settingsXml)
...
+                        // If Uri or assembly qualified type name is repeated, consider data collector as duplicate and ignore it.
+                        continue;
+                    }
+
+                    enabledDataCollectors.Add(settings);
+                }
+            }
+
+            var dataCollectors = enabledDataCollectors.Select(x => new { x.FriendlyName, x.Uri }.ToString());
+            this.requestData.MetricsCollection.Add(TelemetryDataConstants.DataCollectorsEnabled, string.Join(",", dataCollectors.ToArray()));
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs]: Please share a sample value of this telemetry point.

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs [hunk 6]
```diff
@@ -280,5 +294,36 @@ private static string UpdateExtensionsFolderInRunSettings(string settingsXml)
...
+        private void CollectEnabledDataCollectors()
+        {
+            var dataCollectionSettings = XmlRunSettingsUtilities.GetDataCollectionRunSettings(this.settingsXml);
+
+            if (dataCollectionSettings == null || !dataCollectionSettings.IsCollectionEnabled)
+            {
+                return;
+            }
+
+            var enabledDataCollectors = new List<DataCollectorSettings>();
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs]: Do we need to create a new list?

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 7]
```diff
@@ -534,6 +540,66 @@ private static bool IsTelemetryOptedIn()
         }
 
         /// <summary>
+        /// Collect Command Line switches for Telemetry purposes
+        /// </summary>
+        /// <param name="requestData"></param>
+        private void CollectCommands(IRequestData requestData)
```
Comment (harshjain2) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Suggestion : LogCommandsTelemetryPoints

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 8]
```diff
@@ -534,6 +540,66 @@ private static bool IsTelemetryOptedIn()
...
+
+            var frameworkVersionSpecified = this.commandLineOptions.FrameworkVersionSpecified;
+            if (frameworkVersionSpecified)
+            {
+                commandsUsed.Add("/Framework");
+            }
+
+            var settings = this.commandLineOptions.SettingsFile;
+            if (!string.IsNullOrEmpty(settings) && string.Equals(Path.GetExtension(settings), ".runsettings", StringComparison.OrdinalIgnoreCase))
+            {
```
Comment (harshjain2) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: vsmdi file?

Comment (khilan2goel) [reply src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: done

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs (hunk 7 above)
Comment (khilan2goel) [reply src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: done

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs (hunk 6 above)
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs]: Yes. we needed it to see if same data collectors are not added.

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs (hunk 4 above)
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/ProxyDataCollectionManager.cs]: done

**Review by harshjain2 — APPROVED** [review]
1. Validate that DataPoints are coming for MAC and Linux Platforms

File: src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs [hunk 9]
```diff
@@ -160,8 +161,11 @@ public void UpdateLoggerList(string argument, string loggerIdentifier, Dictionar
         /// <summary>
         /// Initializes all the loggers passed by user
         /// </summary>
-        public void InitializeLoggers()
+        /// <param name="requestData">Request Data for Providing Common Services/Data for Discovery and Execution</param>
+        public void InitializeLoggers(IRequestData requestData)
         {
+            var loggersUsed = new List<string>();
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs]: >loggersUsed [](start = 16, length = 11)

We can avoid this, can we use InitializedLoggers instead ?

File: src/vstest.console/app.manifest [hunk 10]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+  <!-- The assemblyIdentity node is needed to fix a binding problem on Win2003 (PS72518) -->
+  <assemblyIdentity version="1.0.0.0" name="MyApplication.app"/>
```
Comment (singhsarab) [inline src/vstest.console/app.manifest]: >MyApplication [](start = 44, length = 13)

Is this correct ?

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 11]
```diff
@@ -536,21 +546,95 @@ private static bool IsTelemetryOptedIn()
         }
 
         /// <summary>
+        /// Log Command Line switches for Telemetry purposes
+        /// </summary>
+        /// <param name="requestData">Request Data providing common discovery/execution services.</param>
+        private void LogCommandsTelemetryPoints(IRequestData requestData)
```
Comment (singhsarab) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Add tests.

**Review by singhsarab — APPROVED** [review]
With suggestions

File: src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs (hunk 9 above)
Comment (khilan2goel) [reply src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs]: Done

File: src/testhost.x86/app.manifest [hunk 12]
```diff
@@ -0,0 +1,27 @@
+﻿<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<!-- Adding same as used in TPv1 Refer: /src/vset/Agile/TestPlatform/RocksteadyCLI/app.manifest -->
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
```
Comment (chakradhar2) [inline src/testhost.x86/app.manifest]: These changes are required for passing the test depending on OS version.

**Pull request merged**